### PR TITLE
feat(specs): update shopify source and task inputs

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1916]

### Changes included:

- Update Shopify source input
- Update Shopify task input

[DI-1916]: https://algolia.atlassian.net/browse/DI-1916?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: specs/ingestion/common/schemas/source.yml [hunk 1]
```diff
@@ -449,7 +450,64 @@ SourceUpdateInput:
...
+    increaseProductCollectionLimit:
+      type: boolean
+      description: |
+        Used to determine if we should index 100 or 200 collections per product.
+        Default is `false`.
+    defaultPriceRatioAsOne:
+      type: boolean
+      description: |
+        The price ratio is determined by `sale_price`/`regular_price`. If no sale price is present, this indexes a 1 instead of a 0.
+        Default is `true`.
```
[the remark this answers is not in the document]
Comment (Fluf22) [reply specs/ingestion/common/schemas/source.yml:478]: We will have to keep an eye on this modification
Are you sure it's valid yml? The linter seems to be confused about the newline+colon

File: specs/ingestion/common/schemas/source.yml [hunk 2]
```diff
@@ -449,7 +450,64 @@ SourceUpdateInput:
...
+      description: |
+        Don't include out-of-stock variants when determining the `max_variant_price` and `min_variant_price` attributes.
+        Default is `true`.
+    includeVariantsInventory:
+      type: boolean
+      description: |
+        Whether we should include an object with the inventory of every variant on the object.
+        Default is `true`.
+    hasCollectionSearchPage:
+      type: boolean
```
[the remark this answers is not in the document]
Comment (Fluf22) [reply specs/ingestion/common/schemas/source.yml:490]: @millotp do you know? I didn't saw anything in the API

File: specs/ingestion/common/schemas/source.yml [hunk 3]
```diff
@@ -449,7 +450,64 @@ SourceUpdateInput:
...
+      description: |
+        Determine if we should index collection information (ID & handle) in the product records.
+    productNamedTags:
+      type: boolean
```
[the remark this answers is not in the document]
Comment (Fluf22) [reply specs/ingestion/common/schemas/source.yml:494]: @millotp to the rescue 😄

[the remark this answers is not in the document]
Comment (millotp) [reply specs/ingestion/common/schemas/source.yml:503]: Screenshot 2024-04-11 at 14 08 59

yes !

File: specs/ingestion/common/schemas/source.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fluf22) [reply specs/ingestion/common/schemas/source.yml]: Perfect

File: specs/ingestion/common/schemas/source.yml [hunk 4]
```diff
@@ -449,7 +450,75 @@ SourceUpdateInput:
...
+      default: true
+      description: |
+        Whether to exclude out-of-stock variants when determining the `max_variant_price` and `min_variant_price` attributes.
+    includeVariantsInventory:
+      type: boolean
+      default: true
+      description: |
+        Whether to include an inventory with every variant for every product record.
+    hasCollectionSearchPage:
+      type: boolean
```
Comment (Fluf22) [inline specs/ingestion/common/schemas/source.yml:499]:
```suggestion
      type: boolean
      default: false
```

File: specs/ingestion/common/schemas/source.yml [hunk 5]
```diff
@@ -449,7 +450,75 @@ SourceUpdateInput:
...
+      description: |
+        Whether to include collection IDs and handles in the product records.
+    productNamedTags:
+      type: boolean
```
Comment (Fluf22) [inline specs/ingestion/common/schemas/source.yml:503]:
```suggestion
      type: boolean
      default: false
```

File: specs/ingestion/common/schemas/task.yml [hunk 6]
```diff
@@ -410,3 +411,54 @@ failureThreshold:
...
+  type: object
+  additionalProperties: false
+  description: Represents the required elements of the task input when using a `shopify` source.
+  properties:
+    metafields:
+      type: array
+      items:
+        $ref: '#/ShopifyMetafield'
+    market:
+      $ref: '#/ShopifyMarket'
```
Comment (millotp) [inline specs/ingestion/common/schemas/task.yml:426]: those 2 props are required

**Review by millotp — COMMENTED** [review]
one last thing

File: specs/ingestion/common/schemas/task.yml (hunk 6 above)
Comment (Fluf22) [inline specs/ingestion/common/schemas/task.yml:426]:
```suggestion
ShopifyInput:
  title: Shopify input
  type: object
  additionalProperties: false
  description: Represents the required elements of the task input when using a `shopify` source.
  properties:
    metafields:
      type: array
      items:
        $ref: '#/ShopifyMetafield'
    market:
      $ref: '#/ShopifyMarket'
  required:
    - metafields
    - market
```

**Review by millotp — APPROVED** [review]
thanks !

**Pull request merged**